File: App/lib/datamodels/ticket_type.dart
class TicketType {
  final String id;
  final String name;
  final String category;
  final double fare;
  final String city;
  final int noLines;
  final String expiry;

  TicketType({
    required this.id,
    required this.name,
    required this.category,
    required this.fare,
    required this.city,
    required this.noLines,
    required this.expiry,
  });

  factory TicketType.fromJSON(Map<String, dynamic> json) {
    return TicketType(
      id: json['id'],
      name: json['name'],
      category: json['category'],
      fare: json['fare'].toDouble(),
      city: json['city'],
      noLines: json['noLines'],
      expiry: json['expiry'],
    );
  }
}

File: api/models/TicketType.go
package models

import (
	"api/db"

	"go.mongodb.org/mongo-driver/bson"
)

type TicketType struct {
  ID string `bson:"id" json:"id"`
  Name string `bson:"name" json:"name"`
  Category string `bson:"category" json:"category"`
  Fare float64 `bson:"fare" json:"fare"`
  City string `bson:"city" json:"city"`
  NoLines int `bson:"noLines" json:"noLines"`
  Expiry string `bson:"expiry" json:"expiry"`
}

func GetTicketTypes(city string) ([]TicketType, error) {
    cursor, err := db.TicketTypes.Find(ctx, bson.M{
      "city": city,
    })

    if err != nil {
      return []TicketType {}, err
    }

    ticketTypes := []TicketType {}

    err = cursor.All(ctx, &ticketTypes)
    if len(ticketTypes) == 0 {
      ticketTypes = []TicketType {}
    }

    return ticketTypes, err
}

func GetTicketType(id string) (TicketType, error) {
  var ticketType TicketType

  err := db.TicketTypes.FindOne(
    ctx, 
    bson.M {
      "id": id,
    },
  ).Decode(&ticketType)

  return ticketType, err
}